### PR TITLE
Generate version type from swagger spec

### PR DESCRIPTION
**Pull request by dnephin**
Part of #27919

### CODE REVIEW

**Comment by dnephin** [comment]
Poule not running on this?

**Comment by LK4D4** [comment]
@dnephin there is gofmt issue.

**Comment by dnephin** [comment]
Thanks, fixed

**Comment by LK4D4** [comment]
@dnephin thanks for moving to proper status. There is issue with generate-swagger now :)

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

File: api/types/system/system_version.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// ----------------------------------------------------------------------------
+// DO NOT EDIT THIS FILE
+// This file was generated by `swagger generate operation`
+//
+// See hack/generate-swagger-api.sh
+// ----------------------------------------------------------------------------
+
+// SystemVersionOKBody system version o k body
+// swagger:model SystemVersionOKBody
+type SystemVersionOKBody struct {
```
Comment (cpuguy83) [inline api/types/system/system_version.go]: This isn't used anywhere?

Comment (dnephin) [inline api/types/system/system_version.go]: Oops, looks like I messed up the rebase. Will fix.

File: api/types/types.go [hunk 2]
```diff
@@ -100,21 +100,6 @@ type Ping struct {
 	Experimental bool
 }
 
-// Version contains response of Engine API:
-// GET "/version"
-type Version struct {
-	Version       string
-	APIVersion    string `json:"ApiVersion"`
-	MinAPIVersion string `json:"MinAPIVersion,omitempty"`
```
Comment (cpuguy83) [inline api/types/types.go]: Are we skipping these `omitempty` settings in the new types? Not sure why they are `omitempty` anyway.

Comment (dnephin) [inline api/types/types.go]: I've been trying to preserve them. I didn't want to make any changes to the spec as part of the transition.

Comment (cpuguy83) [inline api/types/types.go]: Yeah but the new type doesn't have them as omitempty?

Comment (dnephin) [inline api/types/types.go]: Ya, something is wrong here. It's working for other types. I need to investigate.

**Comment by vieux** [comment]
LGTM

**Comment by vdemeester** [comment]
@dnephin is this mergeable as is ?

**Comment by dnephin** [comment]
I don't think the missing `omitempty` will break anything, but it's probably better to wait for a fix.

**Comment by cpuguy83** [comment]
Any news?

**Comment by dnephin** [comment]
I looked into it briefly and I couldn't find the problem. I need more time to dig in to go-swagger

**Comment by dnephin** [comment]
I'm going to close this for now, since I haven't had time to look into the problem

**Pull request closed without merging**